core/test/nuxeo-task-page-provider.test.js

```diff
@@ -0,0 +1,204 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+import '../nuxeo-task-page-provider.js';
+import fakeServer from './nuxeo-mock-client';
+import { fixture, html, waitForEvent } from './test-helpers';
+
+/* eslint-disable no-unused-expressions */
+suite('<nuxeo-task-page-provider>', () => {
+  suite('REST Calls', () => {
+    let server;
+    setup(() => {
+      server = fakeServer.create();
+    });
+
+    teardown(() => {
+      server.restore();
+    });
+
+    suite('Responses and Events', () => {
+      let pageProvider;
+      setup(async () => {
+        pageProvider = await fixture(html`
+          <nuxeo-task-page-provider></nuxeo-task-page-provider>
+        `);
+
+        /*
+         * This workaround is needed due to the 'error' event fired in component.
+         * More Details: https://github.com/webcomponents/webcomponentsjs/issues/138
+         */
+        pageProvider.addEventListener('error', (event) => {
+          event.stopPropagation(); // avoid browser to interpret it.
+        });
+      });
+
+      test('Should fire "error" event when server returns fetch error', async () => {
+        server.rejectWith('get', '/api/v1/task');
+
+        pageProvider.fetch();
+        const event = await waitForEvent(pageProvider, 'error');
+        expect(event).to.exist;
+        expect(event.detail).to.exist.and.to.have.key('error');
+        expect(event.detail.error).to.be.an.instanceof(Error);
```

Comment:
a question/interrogation :) : should we add the two assertions/expectation
```
        expect(event.detail).to.exist.and.to.have.key('error');
        expect(event.detail.error).to.be.an.instanceof(Error);
```
as where we already wait for it `const event = await waitForEvent(pageProvider, 'error');`

Reply:
Indeed, we can remove the first one.
Since we are waiting for the promised to be resolved, if it does not return something the test will fail with a timeout.

Reply:
My previous comment was regarding `expect(event).to.exist;`;

Regarding your point, I'm not aware of a way of checking that the object as a property that is an instance of something. It would be easier to check if `error` was 123: `expect(event.detail).to.exist.and.to.have.property('error', 123);`

In this specific situation I'm not sure how to have one line only. The only way is to don't check that `detail` exist and, if not, the test will still fail. But I'm not sure if it helps from a readability point of view. WDYT?

Reply:
In fact, we may have something like: `expect(event.detail).to.exist.and.to.have.property('error').that.is.an.instanceof(Error);`

Which one would be preferable?

Reply:
thanks for the explanations, as you remove the first line, i think we can keep:
```
        expect(event.detail).to.exist.and.to.have.key('error');
        expect(event.detail.error).to.be.an.instanceof(Error);
```
to give more readability as you suggest (https://github.com/nuxeo/nuxeo-elements/pull/205#discussion_r379007000)